utils/middlewares/hsl-jwt-validation-middleware.ts

```diff
@@ -0,0 +1,142 @@
+  IResponseErrorInternal
+} from "@pagopa/ts-commons/lib/responses";
+
+import { readableReportSimplified } from "@pagopa/ts-commons/lib/reporters";
+import { AuthBearer } from "../../generated/definitions/external/AuthBearer";
+import { IConfig } from "../config";
+import { getHubSpidLoginClient } from "../../clients/hubSpidLogin";
+import { IntrospectSuccessResponse } from "../../generated/definitions/hub-spid-login/IntrospectSuccessResponse";
+
+export interface IHslJwtPayloadExtended extends jwt.JwtPayload {
+  readonly name: string;
+  readonly family_name: string;
+  readonly fiscal_number: FiscalCode;
+}
+
+type IjwtIntrospectionCall = (
+  token: NonEmptyString,
+  config: IConfig
+) => Promise<
+  | IResponseSuccessJson<IntrospectSuccessResponse>
+  | IResponseErrorInternal
+  | IResponseErrorForbiddenNotAuthorized
+>;
+export const introspectionCall: IjwtIntrospectionCall = (
+  token: NonEmptyString,
+  config: IConfig
+) =>
+  pipe(
+    TE.tryCatch(
+      () =>
+        getHubSpidLoginClient(
+          config.HUB_SPID_LOGIN_CLIENT_BASE_URL
+        ).introspectToken({
+          body: {
+            token
+          }
+        }),
+      flow(E.toError, () =>
+        ResponseErrorInternal(`Something gone wrong calling introspection`)
+      )
+    ),
+    TE.chain(
+      flow(
+        TE.fromEither,
+        TE.mapLeft(errors =>
+          ResponseErrorInternal(readableReportSimplified(errors))
+        ),
+        TE.map(response => {
+          switch (response.status) {
+            case 200:
+              return response.value.active
+                ? ResponseSuccessJson(response.value)
+                : ResponseErrorForbiddenNotAuthorized;
+            case 403:
+              return ResponseErrorForbiddenNotAuthorized;
+            default:
+              return ResponseErrorInternal(`Something gone wrong`);
+          }
+        })
+      )
+    ),
+    TE.toUnion
+  )();
+
+export type HslJWTValid = (
+  token: NonEmptyString
+) => TE.TaskEither<Error, IHslJwtPayloadExtended>;
+
+export const hslJwtValidation = (
+  token: NonEmptyString,
+  config: IConfig
+  // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+): HslJWTValid => () =>
+  pipe(
+    getValidateJWT(
+      config.HUB_SPID_LOGIN_JWT_ISSUER,
+      config.HUB_SPID_LOGIN_JWT_PUB_KEY
+    )(token),
+    TE.mapLeft(error => new Error(error.message)),
+    // TE.map(tokenPayload => tokenPayload as IHslJwtPayloadExtended)
+    TE.chain(jwtDecoded =>
+      pipe(
+        TE.tryCatch(
+          () => introspectionCall(token, config),
+          error => new Error(error as string)
+        ),
+        TE.mapLeft(error => new Error(error.message)),
+        TE.chain(introspect => {
+          if (introspect.kind === "IResponseSuccessJson") {
+            return TE.right(jwtDecoded as IHslJwtPayloadExtended);
+          } else {
+            return TE.left(new Error("Introspection failed"));
+          }
+        })

```

Comment:
following the refactor above, we can simplify this pipeline
```suggestion
        introspectionCall(token, config),
        TE.fold(
          _ => TE.left(new Error("Something went wrong")),
          // active is always true if we are in this rail
          _ => TE.right(jwtDecoded as IHslJwtPayloadExtended)
        )
```